### PR TITLE
Decision making guide

### PR DESCRIPTION
**Pull request by afshin**
**Background or context to help others understand the change.**
The incoming governing body pull requests will refer to a common set of decision making strategies, so this pull request represents the decision making guide that all governing bodies within the new model will adopt.

Please consider the draft and offer suggestions.

**A brief summary of the change.**
This adds a decision making guide document.

**What is the reason for this change?**
To create a common "algorithm" for Jupyter sub-projects to use in their decision-making.

## The process ❗

The discussion phase is meant to gather input and multiple perspectives from the community.
Make sure that the community has had an opportunity to weigh in on
the change **before calling a vote**.

cc: @fperez @ellisonbg @tgeorgeux, we decided during the most recent governance call to open this draft PR.

### CODE REVIEW

**Comment by meeseeksmachine** [comment]
This pull request has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/governance-office-hours-meeting-minutes/1480/118

File: decision_making.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+The various Jupyter decision making bodies often have vast scopes and members are often busy. As a result, there are typically a subset of members who have the time, interest, and expertise to dive into the details of a given decision. We believe that open, transparent, and time-bound voting with no quorum will enable these ad-hoc, informal subgroups to form spontaneously on a per-decision basis, in a manner that supports our goals of agility and participation. We have found that formal quorums slow decision making substantially, and don’t actually guarantee true participation.
```
Comment (willingc) [inline decision_making.md:19]: "No quorum" is not a healthy governance practice unless you are coupling it with a specific time frame (7 days) with formal notification of stakeholders that a vote is taking place.

I would completely strike the last sentence as the statements are anecdotal. Not meeting quorums are a red flag that a decision making body is ineffective. Quorums don't guarantee participation though without them the likelihood of a minority railroading through a decision is much higher. Instead of no quorum, it would be more responsible to set a minimum quota of 30%. [For reference from BoardSource](https://boardsource.org/resources/board-meeting-quorum/)

File: decision_making.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## Guidance for decision makers and those participating in decision making
+
+- We encourage teams to always seek to balance the core guiding principles of this process as stated above: balancing agile decision making with an open and collaborative culture of participation from all relevant stakeholders.
+- During the consensus seeking phase, don’t interpret a lack of participation, discussion, or feedback as consensus or the lack thereof. Dissent can be silent and sometimes people are supportive but busy. 
+- If you are interested in a decision being made, it is your responsibility to encourage voting member/stakeholders/community participation in the decision making process. If you can’t get such participation, you may want to hold off on doing any significant work on the matter.
+- We recognize that a no quorum voting scheme could be mistaken for a low participation voting scheme – that is not our goal. To guard against this, teams should always strive for broad participation.
```
Comment (willingc) [inline decision_making.md:28]: Let's strike this entirely. No quorum provides no incentive to strive for broad participation.

File: decision_making.md (hunk 1 above)
Comment (willingc) [reply decision_making.md:19]: Also applying a "no quorum" to a Board of Directors falls short of responsible governance. For example, 1 board member could make decisions on behalf of the entire board. Some level of quorum guards against this possibility. With no quorum, someone with a conflict of interest could make an self-interested (even if unintentional) individual decision that could jeopardize the project.

**Review by willingc — CHANGES_REQUESTED** [review]

File: decision_making.md (hunk 1 above)
Comment (jasongrout) [reply decision_making.md:19]: Furthermore, having some form a quorum requirement is a protection for the board members against being pressured or intimidated for action, even if they would rather not take individual action. Once there is no quorum requirement, an individual can be pressured to take action where they would otherwise want to seek board discussion.

File: decision_making.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+- Special guidelines for software projects making decisions:
+  - Add a link to this document in your issue and PR templates along with a link to the decision making body membership (see recommended language below).
+  - Create a “Consensus Seeking” and “Vote” tags to make it easy to find issues/PRs where a decision is being made.
+  - When a vote is called on GitHub, the sponsor should summarize the decision and paste a checklist of the voting members to use in voting in the top entry (description) of the issue/PR.
+  - When communications (in whatever number of channels) point to a discussion that requires a project-wide decision, and before major implementation work starts, a standalone issue should be opened that summarizes the issue, points to relevant prior discussions, and calls for an explicit vote/decision. This issue should be tightly scoped to the relevant decision only, and include the "Vote" tag.
+  - When an implementation of a decision appears in pull request, link to the relevant “Vote” issue.
+  - This decision making process is incompatible with individuals merging their own PRs in most situations.
```
Comment (jasongrout) [inline decision_making.md:35]: We can distinguish here between individuals reviewing their own PRs and merging their own PRs. In some systems, once other people have signed off on a PR, the original author or anyone else can merge it (for example, it might be signed off conditional on tests passing, and the original author might merge once tests have passed).

Comment (jasongrout) [reply decision_making.md:35]: Also this doesn't account for the somewhat common (at least not uncommon) occurrence of two people working on a PR, each reviewing the other's work, and then one of them merging it.

Comment (afshin) [reply decision_making.md:35]: We should remove this statement because each repository's contributors already have a strong culture of what is acceptable when it comes to merging pull requests.

**Comment by fperez** [comment]
We've made a number of updates to the language, so hopefully the document is clearer when read in isolation  (i.e. outside the context of this PR conversation).  But to try to also answer directly some of the feedback @willingc @jasongrout brought up (thx a lot btw!!):
- There is a 7-day voting period and we added more explicit language about notification. This is all about trying to create a culture of good discussion and communication, where people participate openly and engage with the issues, so we're absolutely not looking to create loopholes that can be used to "submarine" decisions.
- The language about quorum has been updated, to better emphasize that the key principle we want to optimize is engaged participation. We hope this closes avenues for the failure modes that having low/no quorum requirements can engender.
- During some of the live discussions we've somewhat converged on avoiding hard quorum requirements across the entire project (so that e.g. smaller teams and repos can operate more nimbly). Do you think that the updates thus far address these concerns?

(BTW the 'we' above was @afshin @ellisonbg and me during the last meeting :)

**Comment by betatim** [comment]
Thinking out loud about quorum, speed, minimum vote periods:

* 7 days is "not long" when you consider people (frequently) go on holiday/a conference for a week or more
* >7 days is "pretty long" if you are waiting for the result to move forward
* the amount of annoyance/disengagement generated from missing a vote depends on the level at which the vote is happening (boarding level vs team level, missing a lower level one is less annoying than a higher one)
* chances of getting the attention&time of busy people increases the further you schedule with them in advance

This made me wonder if having a fixed cadence for votes would help people to schedule time ("every 3rd Thursday of the month is voting deadline, let me put that in my agenda") and as a result increase the participation? It gives people who are waiting a clear time frame and "plannability" ("We need the board to vote on this so we need to get it to them by X as the next scheduled vote will be too late"). It also signals to would-be decision making body members that if they want to take on this position there is an expectation that they have time to cast a ballot once a month (monthly is just a random cadence I picked for the sake of concreteness).

**Comment by damianavila** [comment]
> This made me wonder if having a fixed cadence for votes would help people to schedule time

This is a really interesting idea. Not sure how that will work in different contexts such as the different decision levels this document is a guide for, but I would bet that could help overall.

> Do you think that the updates thus far address these concerns?

I think the updates make the guide better than the previous iteration, still, I am not totally convinced about not requiring some sort of quorum across the whole project (even if the details for that quorum is different for a different part of the ecosystem at least there is an idea or quorum protecting the decision making process)...

**Comment by tgeorgeux** [comment]
> This made me wonder if having a fixed cadence for votes would help people to schedule time ("every 3rd Thursday of the month is voting deadline, let me put that in my agenda") and as a result increase the participation?

This is an interesting idea, I think if a specific team or decision-making group wanted to go with this cadence they could. We purposely were trying not to be prescriptive behind the 'how' these guidelines are enacted so each team can create a culture that works for them. 

Sometimes lack of participation is because there's a lack of caring, an example being if one or two UX minded people are making color decisions, others may not be as interested. We wanted to give people permission to not care about every decision without stopping progress. 

Sometimes a lack of participation is due to a lack of time, even on decisions that are important. We wanted to give solid minimums for people to have a chance to engage, we also expect there to be differences in how 1-way vs 2-way doors are treated. ie: if the decision is to change a long-held upon protocol, that would be a longer discussion, consensus would be sought, hopefully, achieved, and then the decision would be enacted. But if there is a clear disagreement and consensus is not reached, we ought to be able to vote on a direction and try it. 

In short, I think having some sort of a cadence is a great idea, but I think we should force that cadence on every project.

File: decision_making.md (hunk 1 above)
Comment (willingc) [reply decision_making.md]: Here's a gist that calculates quorum and simple majority on decisions for a group size and quorum level: https://gist.github.com/willingc/dc56d5d11b741e78859e589a59523821 

It's unclear to me what is driving the need for "no quorum". Minimal quorum isn't a heavy burden.

**Comment by meeseeksmachine** [comment]
This pull request has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/governance-office-hours-meeting-minutes/1480/119

File: decision_making.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Decision Making Guide
+
+This document describes how Jupyter’s governing bodies (Board of Directors, Software Steering Council, Software Projects, non-software Working Groups) make decisions. Because individuals often work across multiple decision making groups, all Jupyter governing bodies are expected to follow this approach.
+
+This document provides guidelines but is not meant to be a comprehensive procedure manual. We seek to honor the principles of collaboration, inclusive participation, and responsive decision making. If a situation arises that is not explicitly covered by the guidelines in this document, we encourage all teams to find a path forward with decisions, filling in the gaps in formal decision making procedure with adherence to these principles.
```
Comment (choldgraf) [inline decision_making.md:5]: I find this sentence a bit confusing. When I first read it, I interpreted the rest of the page's content as "these are suggestions but not rules". But then in the next section it says things like `"Formally, these decision making bodies make decisions using consensus seeking with an option to call a vote to move the decision forward."` which seems like something closer to a mandate. Which interpretation is correct?

File: decision_making.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+All official Jupyter projects, working groups, governing bodies, and official initiatives must have a clearly defined, and documented decision making body. Formally, these decision making bodies make decisions using consensus seeking with an option to call a vote to move the decision forward.
+
+Depending on the governing body, decisions may be proposed by members of the decision making body, or by a broader group of stakeholders/contributors.
+
+Decision making, even when it begins with simple informal discussion, starts with consensus seeking. The goal of this discussion phase is to refine the proposal, consider alternatives, weigh trade-offs, and attempt to find informal consensus. The legitimacy of the consensus seeking process is predicated on all stakeholders having their voice heard; teams must be proactive in providing opportunity for all relevant stakeholders to provide input. If the decision making body arrives at informal consensus, they may immediately move to document and enact the decision. This is the consensus seeking phase.
```
Comment (choldgraf) [inline decision_making.md:15]: Just a quick note that for these kinds of sections that lay out different steps of a procedure, I think it is helpful to begin each paragraph with highlighted text or a sub-header that denotes that step's name, so that readers are able to orient themselves to the text more easily

File: decision_making.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+At any time during the consensus seeking phase, any member of the decision making body can call the matter to a vote. When that member (the sponsor) calls the vote, they shall summarize the proposal in its current form and notify the entire decision making body. After the proposal is seconded by another member of the decision making body, members have 7 days to vote. The governing body may consider longer voting periods as necessary for special circumstances, or shorter periods only if all voting members are present. The result will be determined with no quorum by a simple majority for binary decisions and ranked choice for multiclass decisions (one among many, or many among many). The sponsor may update the proposal at any point during the voting period, in which case, the voting period will be reset.
```
Comment (choldgraf) [inline decision_making.md:17]: I feel like there are certain kinds of decisions where you really do want quorum (e.g. a board of directors, a project-wide decision on a JEP, etc) and others where you probably don't need it. Enforcing quorum/no-quorum across all decisions will likely feel inadequate for one or the other type of decision unless we differentiate the two somehow.

Perhaps we can identify an explicit subset of decision-making bodies (e.g. board of directors), or types of decisions (e.g. project-wide JEPs) that require quorum, and for others it is left to the project to decide?

**Review by choldgraf — COMMENTED** [review]
A few quick thoughts - sorry it took me so long to read this but it turns out 👶s require lots of attention 😬

**Comment by choldgraf** [comment]
I made a few comments in the PR - also a quick thought on @betatim's point - it occurs to me that his proposal of a regular cadence for decision-making feels natural with what the JupyterHub project already does. I feel that for most decisions, they are made on-the-fly because a consensus is quickly reached. However for more complex, controversial, etc decisions, we often say "let's discuss at the team meeting". I think it is reasonable that most people who are on the "decision-making body" of a project should also be committed-enough to attend team meetings and such (or cast and absentee vote). So I think this feels reasonable to me.

File: decision_making.md (hunk 6 above)
Comment (willingc) [reply decision_making.md:17]: Governance decisions (BoD and JEPs) benefit from having a quorum. Without quorum and minutes, there is no way to measure the effectiveness of transparency or adequately make sure that folks are aware of a vote.

Decisions that do not need quorum are those that are more along the lines of day-to-day operations of the project.

Comment (damianavila) [reply decision_making.md:17]: I like this idea of quorum/no-quorum for well-defined subsets.

File: decision_making.md [hunk 7]
```diff
@@ -2,40 +2,45 @@
...
-- Do not call a vote to short-circuit an ongoing discussion that is still productive in terms of exploring ideas and feedback. Votes should be called only when discussion has reasonably explored the space and stakeholders have had a voice, to clarify if lack of input is consensus or simply folks being busy, or to record key decisions where an explicit vote will be useful (e.g. to signal an important decision to the broader community).
+- Decision making bodies and those proposing decisions should explicitly distinguish between decisions that are two-way doors (easy to reverse later) and one-way doors (difficult or impossible to reverse later). For one-way doors, teams should carefully weigh alternatives and tradeoffs and take extra care to ensure broad participation and stakeholder input. For two-way doors, teams should feel free to move quickly, without compromising the principles and procedures described herein.
+
+reasonably explored the space and stakeholders have had a voice, to clarify if lack of input is consensus or simply folks being busy, or to record key decisions where an explicit vote will be useful (e.g. to signal an important decision to the broader community).
```
Comment (ivanov) [inline decision_making.md:43]: I think this line was leftover from a previous edit. The gist is captured on lines 32-33, so I suggest removing this, or incorporating it on 32-33 above.

**Comment by meeseeksmachine** [comment]
This pull request has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/governance-office-hours-meeting-minutes/1480/136

File: decision_making.md (hunk 6 above)
Comment (willingc) [reply decision_making.md]: Can we be explicit that BoD and SSC votes should have a quorum? Day-to-day operations in the subproject could be exempt from that. Lack of quorum in BoD and SSC votes leaves me concerned about the governance.

File: decision_making.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
+# Decision Making Guide
+
+This document describes how Jupyter’s governing bodies (Board of Directors, Software Steering Council, Software Projects, non-software Working Groups) make decisions. Because individuals often work across multiple decision making groups, all Jupyter governing bodies are expected to follow this approach.
+
+We seek to honor the principles of collaboration, inclusive participation, and responsive decision making. Some aspects of this decision making guide are required of all teams. Others are provided as recommendations and are optional.
+
+Finally, the Jupyter Board of Directors may either intervene, or be called upon by members of a team, in the event that issues arise in the decision making process. This includes, but is not limited to: process ambiguity, violations of the decision making process, mitigating circumstances that require process exceptions, etc.
```
[the remark this answers is not in the document]
Comment (afshin) [reply decision_making.md:7]: The working groups are going to be in the PR for the Board of Directors, so it probably makes sense to take up this line of inquiry there.

**Comment by afshin** [comment]
The content of this PR has been moved into: https://github.com/jupyter/governance/pull/98
